src/dom-element-getter-helpers.ts

```diff
@@ -8,12 +8,12 @@ type AllProps<ExtraProps = {}> = AppProps &
     appName?: string;
   };
 
-interface HelperOpts {
-  domElementGetter?(): HTMLElement;
+interface HelperOpts<ExtraProps = {}> {
+  domElementGetter?(props?: ExtraProps): HTMLElement;
```

Comment:
Hmm I think this should be more than just the extra props - also the AppProps. This is because the single-spa props get merged with the extra props to form the final props.

```suggestion
  domElementGetter?(props?: ExtraProps & AppProps): HTMLElement;
```

See code below for AppProps.

https://github.com/single-spa/single-spa/blob/e63af429ad676fa2468af08b076ee154ebaa42be/typings/single-spa.d.ts#L12

Looking at the single-spa-react types, I'm not sure it's doing that part correct either, with RootComponentProps not including the sinlge-spa AppProps. So it's possible that updating the code in dom-element-getter-helpers to use AppProps causes issues with single-spa-react, in which case I think I'd prefer trying to fix both places.